library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -0,0 +1,125 @@
+# encoding: utf-8
+
+# ***************************************************************************
+#
+# Copyright (c) 2015 SUSE LLC
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail,
+# you may find current contact information at www.novell.com
+#
+# ***************************************************************************
+# File: fs_snapshot.rb
+#
+# Authors:
+#	Imobach Gonzalez Sosa <igonzalezsosa@suse.com>
+
+require "yast"
+
+module Yast2
+  class SnapperNotConfigured < StandardError
+    def initialize
+      super "Snapper is not configured."
+    end
+  end
+
+  # Class for managing filesystem snapshots. It's important to note that this
+  # class is intended to be used during installation/update so it uses the
+  # Snapper's CLI because the DBus interface is not available at that time.
+  class FsSnapshot
+    FIND_CONFIG_CMD = "/usr/bin/snapper --no-dbus list-configs | grep \"^root \" >/dev/null"
+    CREATE_CONFIG_CMD = "/usr/bin/snapper --no-dbus create-config -f btrfs /"
+    CREATE_SNAPSHOT_CMD = "/usr/lib/snapper/installation-helper --step 5 --description \"%s\""
+    LIST_SNAPSHOTS_CMD = "LANG=en_US.UTF-8 /usr/bin/snapper --no-dbus list"
+
+    attr_reader :number, :snapshot_type, :previous, :timestamp, :user, :cleanup, :description
+
+    # Determines whether snapper is configured or not
+    #
+    # @return [true,false] true if it's configured; false otherwise.
+    def self.configured?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), FIND_CONFIG_CMD)
+      out["exit"] == 0
+    end
+
+    # Configures snapper
+    #
+    # @return [true,false] true if it's configured; false otherwise.
+    def self.configure
+      if configured?
+        true
+      else
+        out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), CREATE_CONFIG_CMD)
+        out["exit"] == 0
+      end
+    end
+
+    # Creates a new snapshot
+    #
+    # It raises and exception if Snapper is not configured.
+    #
+    # @param description [String] Snapshot's description.
+    # @return [FsSnapshot,nil] The created snapshot if the operation was
```

Comment:
NP: I don't see into snapper at all, but it if I understand it correctly it returns a sort of snapshot id. Snapshot and snapshot id has slightly different meaning to me (in case of snapshot I'd expect something else than number).